src/content/plugins/progress-plugin.md

```diff
@@ -0,0 +1,96 @@
+---
+title: ProgressPlugin
+contributors:
+  - elliottsj
+---
+
+The `ProgressPlugin` provides a way to customize how progress is reported during a compilation.
+
+## Usage
+
+Create an instance of `ProgressPlugin` with a handler function which will be called when hooks report progress:
+
+```js
+const handler = (percentage, message, ...args) => {
+  // ...
+};
+
+new webpack.ProgressPlugin(handler);
+```
+
+* `percentage`: a number between 0 and 1 indicating the completion percentage of the compilation.
+* `message`: a short description of the currently-executing hook.
+* `...args`: zero or more additional strings describing the current progress.
+
+## Supported Hooks
+
+The following hooks report progress information to `ProgressPlugin`.
+
+T> _Hooks marked with * allow plugins to report progress information using `reportProgress`. For more, see [Plugin API: Reporting Progress](/api/plugins/#reporting-progress)_
+
+### Compiler
```

Comment:
this will show in the sidebar menu as `Compiler`, maybe dont use `###` here but just make it bold? Because the name of sidebar entry should say `Compiler's hooks which support reporting progress` to make sense but it is too long to fit in. I think the previous heading is enough as an entry for the sidebar (Supported Hooks)

Reply:
Done